tests/venice-pulsar-test/src/pulsarintegrationtest/java/sink/PulsarVeniceSinkTest.java

```diff
@@ -0,0 +1,281 @@
+    // + " --cluster " + clusterName
+    // + " --store " + storeName);
+    //
+    // assertTrue(res.getStdout().contains("\"status\" : \"ONLINE\""));
+    // });
+
+    // await() above does not work, issue number https://github.com/linkedin/venice/issues/387
+    // Have to sleep to let the Venice store become writable,
+    // otherwise Sink's write to venice will not fail but the data won't appear in Venice.
+    Thread.sleep(15000);
+
+    PulsarClient client = PulsarClient.builder()
+        .serviceUrl("pulsar://localhost:6650")
+        .authentication(new AuthenticationDisabled())
+        .allowTlsInsecureConnection(true)
+        .build();
+    Producer<KeyValue<String, Person>> destTopicProducer =
+        client.newProducer(Schema.KeyValue(Schema.STRING, personSchema))
+            .topic("t1/n1/input")
+            .blockIfQueueFull(true)
+            .create();
+
+    final int numRecordsToSend = 100;
+    for (int i = 0; i < numRecordsToSend; i++) {
+      LOGGER.info("Sending {}", i);
+      Person person = new Person("name" + i, i);
+      destTopicProducer.sendAsync(new KeyValue<>(person.getName(), person));
+    }
+
+    LOGGER.info("Flush");
+    destTopicProducer.flush();
+    destTopicProducer.close();
+    client.close();
+
+    LOGGER.info("Querying Venice to check messages");
+    for (int i = 0; i < numRecordsToSend; i++) {
+      String name = "name" + i;
+      int expectedAge = i;
+
+      Awaitility.await().atMost(30, TimeUnit.SECONDS).untilAsserted(() -> {
+        // StdErr will have some noise like
+        // "StatusLogger Log4j2 could not find a logging implementation."
+        ExecResult res = execByService(
+            "venice-client",
+            "bash",
+            "-c",
+            "java -jar /opt/venice/bin/venice-thin-client-all.jar " + storeName + " " + name + " " + veniceRouterUrl
+                + " " + "false" + " " + "\"\"");
+        assertTrue(res.getStdout().contains("key=" + name));
+        assertFalse(res.getStdout().contains("value=null"));
+        assertTrue(res.getStdout().contains("value=" + "{\"age\": " + expectedAge + ", \"name\": \"" + name + "\"}"));
+      });
+    }
+
+    LOGGER.info("Deleting Pulsar Venice Sink");
+    ExecResult deleteSinkRes = execByServiceAsssertNoStdErr(
+        "proxy",
+        "bash",
+        "-c",
+        "/pulsar/bin/pulsar-admin sinks delete --tenant t1 --namespace n1 --name venice");
+    assertTrue(deleteSinkRes.getStdout().contains("Deleted successfully"));
+  }
+
+  private void initVeniceStore(String veniceControllerUrl, String jar, String clusterName, String storeName)
+      throws Exception {
+    execByServiceAsssertNoStdErr(
+        "venice-client",
+        "bash",
+        "-c",
+        "java -jar " + jar + " --empty-push --url " + veniceControllerUrl + " --cluster " + clusterName + " --store "
+            + storeName + " --push-id init --store-size 1000");
```

Comment:
I wonder if this is the issue when the sleep is removed... in theory, this admin command should do the empty push and keep polling until it is successful, but looking at the admin tool code, there are cases where it might short-circuit earlier. I wonder if there may be a situation where the empty push is kicked off, but this command fails to block until it is done.